Frontend/src/components/Form.js

```diff
@@ -134,6 +134,74 @@ const Form = ({ setDisplayCalendar, setTerm, setSchedules, setScheduleCount, set
         });
     };
 
+    const hasDuplicateCourses = (form, term) => {
+        const inputCourses = {};
+        const coursesSubString = new Set([]);
+        // Check for exact matching courses that are duplicates
+        for (const key in form) {
+            if (key.startsWith('course') && form[key].trim() !== '') {
+                const value = form[key].trim();
+                coursesSubString.add(value.split(" ").slice(0, 2).join(" "));
+                if (inputCourses[value]) {
+                    return true;
```

Comment:
This will return true even if the course has multiple sections, not sure if this is intended e.g.:
![image](https://github.com/ConnorMarcus/CarletonClassScheduler/assets/73544123/bc0f35ea-f07a-4993-9584-354b262a23c9)